### PR TITLE
Make this client adapter compatible with PSR18.

### PR DESCRIPTION
**Pull request by Nyholm**
This simple PR (and test fixes) will be the only change to make us compatible with PSR18, right?

#### TODO

- [x] Fix the tests. Make sure we can install php-http/client-integration-tests
- [x] Maybe require PSR-18 explicity
- [x] Show that we `provide` psr-18. 

FYI @fbourigault

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -15,8 +15,8 @@
         }
     ],
     "require": {
-        "php": "^5.5 || ^7.0",
-        "php-http/httplug": "^1.0",
+        "php": "^7.1",
+        "php-http/httplug": "2.x-dev",
```
Comment (GrahamCampbell) [inline composer.json]: `^2.0@dev`?

**Comment by dbu** [comment]
i merged #52, please rebase. 

this will be version 1.2, right? no bc breaks for consumers?

**Comment by Nyholm** [comment]
Since this `Client` is not final, adding a return type hint is considered a BC break, right?

**Comment by fbourigault** [comment]
As this library is just an adapter, PSR-18 should not be directly implemented (i.e. I guess we expect this library to be deprecated some time in the future).

File: composer.json [hunk 2]
```diff
@@ -16,16 +16,17 @@
     ],
     "require": {
         "php": "^7.1",
-        "php-http/httplug": "^1.0",
+        "php-http/httplug": "^2.0@dev",
```
Comment (sagikazarmark) [inline composer.json]: I guess this should be ^2.0

File: composer.json [hunk 3]
```diff
@@ -16,16 +16,17 @@
...
         "guzzlehttp/guzzle": "^6.0"
     },
     "require-dev": {
         "ext-curl": "*",
-        "php-http/client-integration-tests": "^0.6"
+        "php-http/client-integration-tests": "dev-master"
```
Comment (sagikazarmark) [inline composer.json]: Maybe we should tag a new version from this as well?

Comment (Nyholm) [reply composer.json]: Yeah

**Comment by Nyholm** [comment]
I’m happy to merge this.

File: composer.json [hunk 4]
```diff
@@ -16,16 +16,17 @@
     ],
     "require": {
         "php": "^7.1",
-        "php-http/httplug": "^1.0",
```
Comment (joelwurtz) [inline composer.json]: ^1.0|^2.0 does not work ?

**Comment by sagikazarmark** [comment]
I would love to see the integration test package tagged first.

**Comment by xabbuh** [comment]
php-http/guzzle6-adapter#58 seems to support @dbu's reasoning in https://github.com/php-http/guzzle6-adapter/pull/51#pullrequestreview-171434325

**Comment by dbu** [comment]
bumped the branch alias.

do we want to add PSR-18 explicitly to be more clear in composer.json and packagist about what we do?

**Comment by xabbuh** [comment]
> do we want to add PSR-18 explicitly to be more clear in composer.json and packagist about what we do?

I think this makes sense.

**Comment by Nyholm** [comment]
Thank you for the reviews

**Pull request merged**